cluster-autoscaler/core/scale_up.go

```diff
@@ -477,10 +482,11 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 			klog.V(1).Infof("Capping size to max cluster total size (%d)", context.MaxNodesTotal)
 			newNodes = context.MaxNodesTotal - len(nodes) - len(upcomingNodes)
 			context.LogRecorder.Eventf(apiv1.EventTypeWarning, "MaxNodesTotalReached", "Max total nodes in cluster reached: %v", context.MaxNodesTotal)
+			err = errors.NewAutoscalerError(
```

Comment:
Shouldn't that happen inside an if and not outside?

Reply:
Moved it